src/App.js

```diff
@@ -0,0 +1,126 @@
+    orderBy,
+    serverTimestamp,
+    setDoc,
+    doc,
+    deleteDoc,
+} from "firebase/firestore";
+import { useCollectionData } from "react-firebase-hooks/firestore";
+
+const firebaseConfig = {
+    apiKey: "AIzaSyAwgiUzrKROpWbF5MsvzmVZR117tTiYD9s",
+    authDomain: "cs124-lab3-23fc7.firebaseapp.com",
+    projectId: "cs124-lab3-23fc7",
+    storageBucket: "cs124-lab3-23fc7.appspot.com",
+    messagingSenderId: "850754803504",
+    appId: "1:850754803504:web:7fc3629eb6e96acfd61af5"
+};
+
+const firebaseApp = initializeApp(firebaseConfig);
+const db = getFirestore(firebaseApp);
+const collectionName = "todos";
+const priorityLevels = ["c", "b", "a"]; //the low, medium, high display is in PriorityButtons.js
+
+function App() {
+    const qCreated = query(collection(db, collectionName), orderBy("created"));
+    const [todos, loading, error] = useCollectionData(qCreated);
+    const [showAlert, setShowAlert] = useState(false);
+    const [sortByPriority, setSortByPriority] = useState(false);
+
+    const qSortPriority = query(collection(db, collectionName), orderBy("priority"));
+    const [todosPriority, loadingPriority, errorPriority] = useCollectionData(qSortPriority);
+
+    if (loading | loadingPriority) {
+        return "Loading!";
+    }
+
+    if (error | errorPriority) {
+        return error.toString();
+    }
+
+    function handleItemChanged(itemId, field, newValue) {
+        setDoc(doc(db, collectionName, itemId),
+            {[field]: newValue}, {merge: true});
+    }
+
+    function handleCompletedDeleted() {
+        let completedList = todos.filter(item => item.isChecked);
+        completedList.forEach(item => deleteDoc(doc(db, collectionName, item.id)));
+    }
+
+    function handleItemAdded() {
+        let id = generateUniqueID();
+        setDoc(doc(db, collectionName, id), {
+            id: id,
+            isChecked: false,
+            textInput: "",
+            created: serverTimestamp(),
+            priority: priorityLevels[0]
+        });
+    }
+
+    function alertDelete() {
+        handleCompletedDeleted()
+    }
+
+    function toggleModal() {
+        setShowAlert(!showAlert);
+    }
+
+    function handleSortPriority() {
+        setSortByPriority(!sortByPriority);
+    }
+
+    return <>
+        <h1>To Do List</h1>
+        {todos.length === 0 && <h4>No items</h4>}
+        <div className = "sortButton">
+            {todos.length > 1 && <button type={"button"} onClick={handleSortPriority} id = {"thisButton"}>
+                {sortByPriority ? "Sort by Date Created" : "Sort by Priority"}
+            </button>}
+        </div>
+        <TaskList
+            todo={sortByPriority ? todosPriority.filter(item => !item.isChecked) : todos.filter(item => !item.isChecked)}
+            isCompletedList={false}
+            onItemChanged={handleItemChanged}
+            priorityLevels={priorityLevels}
+        />
+        {todos.filter(item => item.isChecked).length > 0 && <h4>Completed</h4>}
```

Comment:
You do a filter here and then again on line 103. Better to evaluate it once into a local variable and then use that local variable (Don't Repeat Yourself).